### ISSUE TITLE
Vesktop does not recognize Steam games (arrpc enabled)

### ISSUE THREAD

**Issue by diimon04** [OpenAsar/arrpc#105]
When I play games like Stardew Valley (with Steam), my game activity is not displayed on Vesktop. It works with games like Minecraft. Im on Arch Linux with KDE.

**Comment by ToRRent1812** [comment]
I have encountered similar issue

**Comment by DerCommander323** [comment]
Same for me, Arch linux but with Hyprland. I even tried https://github.com/OpenAsar/arrpc/pull/92, but it still doesn't work at all. Directly integrating software (for example VSCodium with a discord rpc plugin) works as intended.

**Comment by OreoStepswitcher** [comment]
It's solved by starting the arRPC server manually. Still an issue though, because Vesktop starts it's own arRPC server, but doesn't recognize it

**Comment by diimon04** [comment]
> Flatpak doesn't support process detection.

sorry for not specifying, but im using the aur version (if it is somehow relevant to the issue)

**Comment by bombasticman** [comment]
I have the same problem on Linux Mint. Games run trough Lutris are fine but my steam games are not detected.

**Comment by BreakingTV** [comment]
bump.
Vesktop has completly removed adding your own game activy on your own because of beeing just a web implementation...

**Comment by mewostick** [comment]
It is worth noting that as a workaround one may host arrpc themselves and enable WebRPC plugin in the settings.

**Comment by DerCommander323** [comment]
> It is worth noting that as a workaround one may host arrpc themselves and enable WebRPC plugin in the settings.

That's what I'm doing (how else is it even possible?) and it's not working.

**Comment by mewostick** [comment]
> (how else is it even possible?)
@DerCommander323 What do you mean here 🤨

**Comment by DerCommander323** [comment]
> @DerCommander323 What do you mean here 🤨

I mean how else are you supposed to set up arRPC?

**Comment by BreakingTV** [comment]
> > @DerCommander323 What do you mean here 🤨
> 
> I mean how else are you supposed to set up arRPC?

In theory it should create its own arRPC instance in the client itself

**Comment by mewostick** [comment]
> In theory it should create its own arRPC instance in the client itself

Vesktop seems to have some sort of limitation that prevents it from scanning for processes it seems (it wasn't the case before though and I wonder why)... It only picks up anything that "emits" a Discord Rich Presence with all of the extra info.

I host arRPC on my own with Node 20 LTS from Homebrew on Linux and it picks up my games normally.

**Comment by BreakingTV** [comment]
Maybe because of the new rich presence Feature and an old Version of arRPC...
This problem started to occure for me after they started working on the new RPC

**Comment by Arcitec** [comment]
Well, I have now tried with the latest external arRPC:

- Vesktop: Settings: "Vesktop Settings": Disable "Rich Presence" (arRPC).
- Fully shut down Vesktop to unregister its `discord-ipc-0` pipe.
- Start external arRPC manually with Node: `node src` in the arRPC directory. It should say `listening at /run/user/1000/discord-ipc-0`.
- Start Vesktop.
- Vesktop: Settings: "Plugins": Enable "WebRichPresence (arRPC)". It should pop up the message "Connected to arRPC" briefly on screen.
- Now start a game. I played a Steam game with Proton. The arRPC terminal window said `[arRPC > process] detected game! Grounded`.

However, in Vesktop, I still don't become marked as playing that game. So maybe something is bugged in Vencord/Vesktop itself.

PS: Vesktop already includes the newest arRPC so the theory people had above that it's "outdated" is not true.

**Comment by mewostick** [comment]
> Edit: Could it be because Discord has changed something in its expected RPC protocol and arRPC uses the old protocol, so it no longer works at all? Maybe.

Could you check if it works for you on Discord in a web browser? Just make sure you don't have any tracker blockers on as they may prevent access to localhost:{PORT} which arRPC needs.

Perhaps I haven't been affected by some A/B changes..

**Comment by Arcitec** [comment]
## I have solved it:

- Be sure that arRPC is enabled inside Vesktop: Settings: "Vesktop Settings": Enable "Rich Presence" (arRPC).
- Then go to Vesktop: Settings: "Plugins": Enable "GameActivityToggle", and restart Vesktop when it asks you to do that.
- Now you will have a game activity icon next to the microphone. It defaults to disabled:

![image](https://github.com/user-attachments/assets/7f58c2a8-683b-498b-a265-681f18f61ad5)

- Click that icon to enable it:

![image](https://github.com/user-attachments/assets/c55d664c-901f-47c2-b85d-e25d1f34fb91)


- Play a game!

![image](https://github.com/user-attachments/assets/2914e9f9-3575-403a-acd3-58f5219005fe)

---

I don't know why Vesktop decided to disable Game Activity / Rich Presence by default and hide its setting inside a plugin. Probably an oversight and "Game Activity" was probably supposed to be enabled by default. I will report it to Vesktop.

**Comment by BreakingTV** [comment]
It doesnt need to be a discord authorized game.
You can use Jetbrains IDEs discord rich presence plugin and works just fine.
I think it is often a problem of Implementation too...

**Comment by Arcitec** [comment]
@BreakingTV Yes, native Linux games and apps that talk to Discord RPC via `discord-ipc` pipe will automatically work and will set the activity.

Windows games that contain built-in support for Discord RPC always need `rpc-bridge` as my post mentioned, to bridge the divide between Windows and Linux socket locations.

**Comment by mewostick** [comment]
> * Be sure that you use native Vesktop. **Not Flatpak.** Because containers (Flatpaks) cannot scan processes on the host, and has lots of other limitations related to Discord RPC.

If I remember correctly, this didn't really work for me on Nobara 40 with a native Vesktop package.

But now on an Ubuntu 24.10 system it works perfectly. I really wonder if some Linux distributions have some kind of a change that causes Vesktop's ArRPC to half-work..

**Comment by zastrixarundell** [comment]
Are you 100% it doesn't work? 

I'm running fedora kinoite (immutable KDE OS). arrpc is a userspace systemd service. Steam and Vesktop are flatpaks and it just works. Game activity it picked up from Steam.

**Comment by cebxan** [comment]
> Are you 100% it doesn't work?
> 
> I'm running fedora kinoite (immutable KDE OS). arrpc is a userspace systemd service. Steam and Vesktop are flatpaks and it just works. Game activity it picked up from Steam.
> 
> ![image](https://private-user-images.githubusercontent.com/30195425/375274499-d4c36be6-8af2-493a-bb67-768a84301b67.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAzMjQ2NzksIm5iZiI6MTczMDMyNDM3OSwicGF0aCI6Ii8zMDE5NTQyNS8zNzUyNzQ0OTktZDRjMzZiZTYtOGFmMi00OTNhLWJiNjctNzY4YTg0MzAxYjY3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDMwVDIxMzkzOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI3MGVmYWVjODQ1MWNmNTA3YTA1ODVkNmM0OGNmZGMwMDMyODg3OTY0MDhiMTZkZWFlMDViYTVhZWM0MGZkNGImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.UfJqgsherUC6HhXqwQ_ysbCWHdIRDfzUPoKytJ0Dod8)

Can you please share with us if you did any special setup to achieve that? Did you set up aarpc separately from Vesktop? Any extra symlinks?

**Comment by Arcitec** [comment]
@cebxan Yes he said that `arrpc is a userspace systemd service` which means it's separate from Vesktop.

He is running arrpc as a permanent user-level background service.

To connect to it, Vesktop must be configured to disable its own built-in arrRPC plugin, and to connect to the local instance via localhost networking port (if I remember correctly).

Hope these hints help you.

But Vesktop inside a Flatpak is still not a good thing. It will still be sandboxed in a way that prevents other apps from talking to its RPC service. And the Vesktop Flatpak has maaaany bugs related to RPC (I described the bugs on their Flathub repo in a ticket). I use the native Vesktop client because it's far superior (built-in arRPC, Discord RPC just works).

**Comment by cebxan** [comment]
@Arcitec Thanks for pointing that out. I wasn't aware that we could run the arrpc service separately from Vesktop and that confused me.

That's really the key to make the Vesktop Flatpak work with RPC. The separate and non-sandboxed arrpc is able to spy the process list and correctly detect running games and other apps. It still has issues with apps like Tidal-Hifi that look for `discord-ipc-0` specifically within a Discord folder (`$XDG_RUNTIME_DIR/app/com.discordapp.Discord/`)

While I agree that the native version of Vesktop works better, the Flatpak is basically a requirement for Atomic Desktops. I'm running Bazzite now and doing my best to understand these limitations and make it work. Sadly there's no way yet to allow a sandboxed app to read the process list.

**Comment by salamijason** [comment]
I face this issue as well. On pikaOS, vesktop native, arrpc server hosted according to instructions on the arrpc page, everything that should be enabled is enabled. most games get detected by arrpc but there are a couple exceptions. the one i've encountered the most is stardew valley (via steam).

**Comment by Edwardl256** [comment]
I'd like to point out that I had to use the Flatpak version in order to share sounds with screensharing. Little did I know that I would be sacrificing game activity.

**Comment by PANICBUTT0N** [comment]
Is there a way to manually add entries to vesktop RPC like you can on Windows?

**Comment by electroniumez** [comment]
for me, enlisted and CS2 doesn't recognize by native arrpc of vesktop (deb)
steam installed with deb file too
